tests/linux/test_platform_linux.cpp

```diff
@@ -554,4 +555,172 @@ INSTANTIATE_TEST_SUITE_P(
+                         {"PRETTY_NAME=\"Ubuntu 21.04\""},
+                         {"VERSION_ID=\"\""},
+                         {"HOME_URL=\"https://www.ubuntu.com/\""},
+                         {"SUPPORT_URL=\"https://help.ubuntu.com/\""},
+                         {"BUG_REPORT_URL=\"https://bugs.launchpad.net/ubuntu/\""},
+                         {"PRIVACY_POLICY_URL=\"https://www.ubuntu.com/legal/terms-and-policies/privacy-policy\""},
+                         {"VERSION_CODENAME=hirsute"},
+                         {"UBUNTU_CODENAME=hirsute"}};
+
+    auto expected = std::pair<std::string, std::string>("unknown", "unknown");
+
+    auto output = mp::platform::parse_os_release(input);
+
+    EXPECT_EQ(expected.first, output.first.toStdString());
+    EXPECT_EQ(expected.second, output.second.toStdString());
+}
+
+TEST_F(PlatformLinux, parse_os_release_single_char_fields)
+{
+    QStringList input = {{"NAME=\"A\""},
+                         {"VERSION=\"21.04 (Hirsute Hippo)\""},
+                         {"ID=ubuntu"},
+                         {"ID_LIKE=debian"},
+                         {"PRETTY_NAME=\"Ubuntu 21.04\""},
+                         {"VERSION_ID=\"B\""},
+                         {"HOME_URL=\"https://www.ubuntu.com/\""},
+                         {"SUPPORT_URL=\"https://help.ubuntu.com/\""},
+                         {"BUG_REPORT_URL=\"https://bugs.launchpad.net/ubuntu/\""},
+                         {"PRIVACY_POLICY_URL=\"https://www.ubuntu.com/legal/terms-and-policies/privacy-policy\""},
+                         {"VERSION_CODENAME=hirsute"},
+                         {"UBUNTU_CODENAME=hirsute"}};
+
+    auto expected = std::pair<std::string, std::string>("A", "B");
+
+    auto output = mp::platform::parse_os_release(input);
+
+    EXPECT_EQ(expected.first, output.first.toStdString());
+    EXPECT_EQ(expected.second, output.second.toStdString());
+}
+
+TEST_F(PlatformLinux, parse_os_release_ubuntu2104lts)
+{
+    QStringList input = {{"NAME=\"Ubuntu\""},
+                         {"VERSION=\"21.04 (Hirsute Hippo)\""},
+                         {"ID=ubuntu"},
+                         {"ID_LIKE=debian"},
+                         {"PRETTY_NAME=\"Ubuntu 21.04\""},
+                         {"VERSION_ID=\"21.04\""},
+                         {"HOME_URL=\"https://www.ubuntu.com/\""},
+                         {"SUPPORT_URL=\"https://help.ubuntu.com/\""},
+                         {"BUG_REPORT_URL=\"https://bugs.launchpad.net/ubuntu/\""},
+                         {"PRIVACY_POLICY_URL=\"https://www.ubuntu.com/legal/terms-and-policies/privacy-policy\""},
+                         {"VERSION_CODENAME=hirsute"},
+                         {"UBUNTU_CODENAME=hirsute"}};
+
+    auto expected = std::pair<std::string, std::string>("Ubuntu", "21.04");
+
+    auto output = mp::platform::parse_os_release(input);
+
+    EXPECT_EQ(expected.first, output.first.toStdString());
+    EXPECT_EQ(expected.second, output.second.toStdString());
+}
```

Comment:
These last three tests all repeat the same structure and could be parameterized. The parameter would be the pair you now set in `expected`, which you could render into `input` with [`QString::arg`](https://doc.qt.io/qt-5/qstring.html#arg). You can look for `TEST_P` in our code to see examples of how that works.

I guess the tests for the empty file and different order are fine as separate tests.